src/rime/core_module.cc

```diff
@@ -20,13 +20,27 @@ static void rime_core_initialize() {
   LOG(INFO) << "registering core components.";
   Registry& r = Registry::instance();
 
-  auto config = new ConfigComponent;
-  config->InstallPlugin(new AutoPatchConfigPlugin);
-  config->InstallPlugin(new DefaultConfigPlugin);
-  config->InstallPlugin(new LegacyPresetConfigPlugin);
-  config->InstallPlugin(new LegacyDictionaryConfigPlugin);
-  r.Register("config", config);
-  r.Register("schema", new SchemaComponent(config));
+  const ResourceType build_output{"config", "build/", ".yaml"};
+  auto config_builder = new ConfigComponent<ConfigBuilder>(
+      [&](ConfigBuilder* builder) {
+        builder->InstallPlugin(new AutoPatchConfigPlugin);
+        builder->InstallPlugin(new DefaultConfigPlugin);
+        builder->InstallPlugin(new LegacyPresetConfigPlugin);
+        builder->InstallPlugin(new LegacyDictionaryConfigPlugin);
+        builder->InstallPlugin(new BuildInfoPlugin);
+        builder->InstallPlugin(new SaveOutputPlugin(build_output));
+      });
+  r.Register("config_builder", config_builder);
+
+  auto config_loader = new ConfigComponent<ConfigLoader>(build_output);
+  r.Register("config", config_loader);
```

Comment:
can we instead use a `ConfigComponent<ConfigBuilder>` or fallback to a builder, to make it more robust by doing timely rebuild if `build/*.yaml` is missing or obsolete?
prior to this change, new edits to YAML files can take effect when switching to a different schema.

Reply:
The improved config is too flexible to easily manage, like when used as a table's dependencies.
So I deliberately separate the builder from run-time `config` component.
It allows
- creating a `build/` on computer for mobile frontend
- code for `ConfigCompiler` to link to a separate deployer executable, not to the run-time
- in future, replacing the built YAML with another format, and remove yaml-cpp as a dependency to the run-time

As a side effect, the change forces user to *deploy* their modifications, and prevents mis-use.

Reply:
From [customization guide](https://github.com/rime/home/wiki/CustomizationGuide), quote
> 如果下載、自己製作了非預設的輸入方案，將源文件複製到「用戶資料夾」後，也用上面的方法將方案標識加入選單！

unquote. However, this is impossible now since config files in user data directory are no longer being built.

Then, where should users place their non-preset schemata? If users can make use of user data directory, they will not be able to sync non-preset schemata across devices.

Reply:
Let me clarify: config_builder, used in the deployment process, reads source YAML files from user|shared data directory and saves compiled YAML in build/. You can treat files under build/ as a cache for processed user data; it's not user facing.
If a mobile client cannot deploy data on device due to hardware or OS limitations, then we can deploy on a computer and transfer build/ to the mobile device. Of course, user has to redo deployment on the computer after adding new files.

Reply:
In actual usage with weasel, source YAML files in user data directory are recognized but not built inside build/ directory, therefore not working. Only source files in shared data directory are built.